crawl-ref/source/webserver/game_data/static/action_panel.js

```diff
@@ -157,26 +157,52 @@ function ($, comm, client, cr, enums, options, player, icons, gui, main,
         }
         $tooltip.css({top: y + 10 + "px",
                      left: x + 10 + "px"});
-        if (slot == -2)
-        {
-            $tooltip.html("<span>Left click: minimize</span><br />"
+
+        switch (slot + NUM_RESERVED_BUTTONS) {
```

Comment:
I would also prefer something a lot more general here, though the exact form I'm not as sure of. One consideration is that it may be good to allow players to hide these extra buttons.

In some cases I'm not entirely sure of the rationale for why these and not other possibilities, e.g. why show spell library rather than cast. Probably the ability menu is another candidate for this. (In the long run, probably what would be ideal is allowing arbitrary commands to appear in this menu.) Some motivation in commit messages would definitely help.